### PR TITLE
Remove default values and comments from mbedtls.doxyfile

### PR DESCRIPTION
**Pull request by AndrzejKurek**
Used the 1.8.17 generated version as a base for comparison with mbedtls.doxyfile. Fields that did not have the default 1.8.17 value:
```
PROJECT_NAME           = "mbed TLS v3.1.0"
OUTPUT_DIRECTORY       = ../apidoc/
FULL_PATH_NAMES        = NO
TAB_SIZE               = 8
OPTIMIZE_OUTPUT_FOR_C  = YES
EXTRACT_ALL            = YES
EXTRACT_PRIVATE        = YES
EXTRACT_STATIC         = YES
EXTRACT_LOCAL_METHODS  = YES
EXTRACT_ANON_NSPACES   = YES
CASE_SENSE_NAMES       = NO
INPUT                  = ../include input
FILE_PATTERNS          = *.h
RECURSIVE              = YES
EXCLUDE_SYMLINKS       = YES
SOURCE_BROWSER         = YES
REFERENCED_BY_RELATION = YES
REFERENCES_RELATION    = YES
ALPHABETICAL_INDEX     = NO
HTML_OUTPUT            = .
HTML_TIMESTAMP         = YES
MATHJAX_RELPATH        = http://cdn.mathjax.org/mathjax/latest
SEARCHENGINE           = NO
GENERATE_LATEX         = NO
LATEX_CMD_NAME         = latex
COMPACT_LATEX          = YES
PAPER_TYPE             = a4wide
MACRO_EXPANSION        = YES
EXPAND_ONLY_PREDEF     = YES
INCLUDE_PATH           = ../include
PREDEFINED             = WIN32 \
                         NTLM \
                         USE_LZO \
                         ENABLE_FRAGMENT \
                         P2MP \
                         P2MP_SERVER \
                         USE_CRYPTO \
                         USE_SSL \
                         ENABLE_PLUGIN \
                         ENABLE_MANAGEMENT \
                         ENABLE_OCC \
                         HAVE_GETTIMEOFDAY
EXPAND_AS_DEFINED      = MBEDTLS_PRIVATE
PERL_PATH              = /usr/bin/perl
CLASS_DIAGRAMS         = NO
MSCGEN_PATH            =
HAVE_DOT               = YES
DOT_GRAPH_MAX_NODES    = 200
MAX_DOT_GRAPH_DEPTH    = 1000
DOT_TRANSPARENT        = YES
```
`MATHJAX_RELPATH` and `LATEX_CMD_NAME` seemed to have changed their default value, `PERL_PATH` and `MSCGEN_PATH` were obsolete (and doxygen complained about that), so I removed them. The rest can be seen in this PR.
Please note that if we have any scripts that search for a given symbol to change its value, and it had its default value in the first place - these scripts will fail.
Related PRs: #5367, #5371
Backport: #5543
Partially fixing Issue #3497

### CODE REVIEW

**Comment by AndrzejKurek** [comment]
Two remaining questions:
- Does this PR requires a changelog entry?
- Do we want to use the proposed 1.8.17 version of doxyfile as a comaprison and do we want to bump doxygen version in our docs and scripts?

**Review by mprse — COMMENTED** [review]
I tried to go through all used and default config options and compare with https://www.doxygen.nl/manual/config.html. And it seems that our config is consistent with documentation. I only found one option which is not in documentation.

I was also wondering why description is also removed for used config options?

**Comment by AndrzejKurek** [comment]
> I tried to go through all used and default config options and compare with https://www.doxygen.nl/manual/config.html. And it seems that our config is consistent with documentation. I only found one option which is not in documentation.
> 
> I was also wondering why description is also removed for used config options?

I assumed that we want to minimize the config to make it easier to find and read what options don't have their default value. Every option can be found in the doxygen documentation anyway, and the names are pretty descriptive.

File: doxygen/mbedtls.doxyfile [hunk 1]
```diff
@@ -1623,289 +39,9 @@ PREDEFINED             = WIN32 \
...
-# Configuration options related to the dot tool
-#---------------------------------------------------------------------------
-
-# If the CLASS_DIAGRAMS tag is set to YES (the default) Doxygen will
-# generate a inheritance diagram (in HTML, RTF and LaTeX) for classes with base
-# or super classes. Setting the tag to NO turns the diagrams off. Note that
-# this option also works with HAVE_DOT disabled, but it is recommended to
-# install and use dot, since it yields more powerful graphs.
-
 CLASS_DIAGRAMS         = NO
```
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply doxygen/mbedtls.doxyfile:43]: [It became deprecated recently](https://www.doxygen.nl/manual/changelog.html)

**Comment by gilles-peskine-arm** [comment]
> Please note that if we have any scripts that search for a given symbol to change its value, and it had its default value in the first place - these scripts will fail.

We don't. Users concievably might if they're integrating Mbed TLS documentation into a bigger or modified project (e.g. Mbed TLS + some drivers in a platform BSP), but more likely I'd expect them to either have their own doxyfile or use ours unmodified.

As a consequence, I don't think this requires a changelog entry.

File: doxygen/mbedtls.doxyfile [hunk 2]
```diff
@@ -1,1616 +1,32 @@
...
-# generate index for LaTeX. If left blank `makeindex' will be used as the
-# default command name.
-
-MAKEINDEX_CMD_NAME     = makeindex
-
-# If the COMPACT_LATEX tag is set to YES Doxygen generates more compact
-# LaTeX documents. This may be useful for small projects and may help to
-# save some trees in general.
-
 COMPACT_LATEX          = YES
```
Comment (gilles-peskine-arm) [inline doxygen/mbedtls.doxyfile:25]: `COMPACT_LATEX` and `PAPER_TYPE` are not relevant since we don't generate latex. If a user wants to tweak the file to generate latex, they can tweak latex settings as well.

File: doxygen/mbedtls.doxyfile [hunk 3]
```diff
@@ -1,1616 +1,32 @@
...
-
-# The PREDEFINED tag can be used to specify one or more macro names that
-# are defined before the preprocessor is started (similar to the -D option of
-# gcc). The argument of the tag is a list of macros of the form: name
-# or name=definition (no spaces). If the definition and the = are
-# omitted =1 is assumed. To prevent a macro definition from being
-# undefined via #undef or recursively expanded use the := operator
-# instead of the = operator.
-
 PREDEFINED             = WIN32 \
```
Comment (gilles-peskine-arm) [inline doxygen/mbedtls.doxyfile:30]: I don't think any of these macro names have had any effect since at least Mbed TLS 2.0, when all identifiers used by the library acquired an `MBEDTLS_` prefix. And I think many were actually obsolete way before, this list hadn't kept up with changes in PolarSSL 0.x. How about removing them, assuming this doesn't change the output (I haven't checked)?

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
I think we should remove the comment with a Doxygen version number since it's meaningless. And we might as well remove options that don't have any effect.

File: doxygen/mbedtls.doxyfile (hunk 3 above)
Comment (AndrzejKurek) [reply doxygen/mbedtls.doxyfile:30]: After turning off the `HTML_TIMESTAMP` to get consistent and reproducible output `sha256sum <(find . -type f -exec sha256sum {} \; | sort)` yields the same results with and without the predefined option. I'm removing them.

**Review by gilles-peskine-arm — APPROVED** [review]

**Comment by AndrzejKurek** [comment]
merge test failure:

> java.io.IOException: Remote call on windows.eu-west-1a (i-076769e3d9c7fd81d) failed

**Review by mprse — APPROVED** [review]

**Comment by mpg** [comment]
I think this need a backport to 2.28, as the other PRs in this series seem to be backported (eg #5367) or waiting for backports (eg #5371) as well. So, I'm labeling "needs: backports" - if you think I'm wrong, please explain why and remove the label.

**Comment by AndrzejKurek** [comment]
> I think this need a backport to 2.28, as the other PRs in this series seem to be backported (eg #5367) or waiting for backports (eg #5371) as well. So, I'm labeling "needs: backports" - if you think I'm wrong, please explain why and remove the label.

I was just thinking that it's not a bug fix, but an improvement that's unnecessary in an old branch that will not have the doxygen version updated.

**Comment by gilles-peskine-arm** [comment]
> I was just thinking that it's not a bug fix, but an improvement that's unnecessary in an old branch that will not have the doxygen version updated.

We don't ship with a doxygen version. People who are using the LTS may still want to upgrade their build tools. E.g. IIRC the LTS will still be around after Ubuntu no longer has a supported release with a doxygen that builds 2.28 without warnings.

**Comment by AndrzejKurek** [comment]
Added backport here: https://github.com/ARMmbed/mbedtls/pull/5543
